src/components/icons/LottieAnimation.tsx

```diff
@@ -1,36 +1,128 @@
 import Player, { LottieRefCurrentProps } from 'lottie-react'
-import React, { useEffect, useRef } from 'react'
+import React, { useCallback, useEffect, useRef, useState } from 'react'
 
 interface LottieAnimationProps {
   animationData: any
   speed?: number
+  color?: string
   onComplete?: () => void
   style?: React.CSSProperties
 }
 
 /**
  * LottieAnimation Component.
  *
- * This component renders a Lottie animation using the `lottie-react` Player.
- * It accepts animation data and an optional speed prop to control the animation speed.
+ * This component utilizes the 'lottie-react' Player to render a Lottie animation.
+ * It accepts animation data and supports customizable speed and color properties,
+ * allowing for dynamic control over the animation's playback rate and the stroke
+ * and fill colors of its elements.
  *
+ * Props:
+ * animationData: The JSON data that defines the Lottie animation.
+ * Speed: Optional. A number that specifies the playback speed of the animation. Defaults to 1 for normal speed.
+ * Color: Optional. A string representing the hex color code used to update the stroke and fill colors of the animation. Defaults to '#FFFFFF' (white) if not specified.
+ * OnComplete: Optional. A callback function that is triggered when the animation completes its playback.
+ * Style: Optional. A React.CSSProperties object to apply inline styles to the player container.
+ * This does not affect the Lottie animation's JSON content or alter its visual elements.
  */
 const LottieAnimation: React.FC<LottieAnimationProps> = ({
   animationData,
-  speed = 1, // Default speed set to 1x
+  speed = 1,
+  color = '#FFFFFF',
   onComplete,
   style,
 }) => {
+  const [key, setKey] = useState(0)
   const lottieRef = useRef<LottieRefCurrentProps | null>(null)
 
+  /**
+   * Converts hex color to RGBA.
+   *
+   * @param hex - The hex color code.
+   * @returns The RGBA values.
+   */
+  const hexToRGBA = useCallback((hex: string): number[] => {
```

Comment:
Since this is a pure function, we should be able to declare it as a regular function outside of the component rather than using `useCallback` for it.